batch/batch/worker/worker.py

```diff
@@ -1664,6 +1665,7 @@ async def run(self):
                     async with self.worker.jar_download_locks[self.revision]:
                         local_jar_location = f'/hail-jars/{self.revision}.jar'
                         if not os.path.isfile(local_jar_location):
+                            self.verify_is_acceptable_query_jar_url(self.jar_url)
```

Comment:
We had a bug in DockerJob here by checking only whether the file exists and not explicitly storing whether the file finished downloading successfully.